arangod/RestHandler/RestEdgesHandler.cpp

```diff
@@ -141,17 +91,46 @@ bool RestEdgesHandler::parseDirection(TRI_edge_direction_e& direction) {
   return true;
 }
 
+namespace {
+std::string queryString(TRI_edge_direction_e dir) {
+  switch (dir) {
+    case TRI_EDGE_IN:
+      return "FOR e IN @@collection FILTER e._to == @vertex RETURN e";
+    case TRI_EDGE_OUT:
+      return "FOR e IN @@collection FILTER e._from == @vertex RETURN e";
+    case TRI_EDGE_ANY:
+      return "FOR e IN @@collection FILTER (e._from == @vertex || e._to == "
+             "@vertex) RETURN e";
+  }
+}
+
+aql::QueryResult queryEdges(TRI_vocbase_t& vocbase, std::string const& cname,
+                            TRI_edge_direction_e dir, std::string const& vertexId) {
+  auto bindParameters = std::make_shared<VPackBuilder>();
+  bindParameters->openObject();
+  bindParameters->add("@collection", VPackValue(cname));
+  bindParameters->add("vertex", VPackValue(vertexId));
+  bindParameters->close();
+  auto options = std::make_shared<VPackBuilder>();
+
+  arangodb::aql::Query query(false, vocbase, aql::QueryString(queryString(dir)),
```

Comment:
This method is not using `RestVocBaseHandler::createTransaction()` anymore as the previous version did.
Wondering if we can still use the edges functionality from inside a cluster transaction.
IIRC then aql::Query can't pick up an existing cluster transaction, so these operations now are run outside of existing cluster transactions, even if the transaction id header is passed to the RestHandler.

I think we should also add tests to verify the transaction behavior and ensure it doesn't break over time.

Reply:
this handler never supported el-cheapo. if the docs state it otherwise the docs are lying

Reply:
No, it wasn't documented. But is there a problem with supporting transactions here?

Reply:
yes

Reply:
no need to improve APIs that should be deprecated anyway